app/controllers/shopify_app/sessions_controller.rb

```diff
@@ -12,7 +12,11 @@ class SessionsController < ActionController::Base
     end
 
     def new
-      authenticate if sanitized_shop_name.present?
+      if params[:reauthorize].present?
+        flash[:error] = I18n.t("reauth_needed")
```

Comment:
I'm not sure I understand the idea here - will we just show a message showing re-auth is needed, rather than actually trigger it?